### PR TITLE
feat: Tabs for Multiple Files 

### LINKED ISSUE #72: Tabs for Multiple Files
Right now if you add multiple files to a snippet, it just shows the one after the other.

Why not do it elegantly with tabs?

### PR DESCRIPTION
**Pull request by Andrei0872**
Closes #72 

Whether updating or creating a note, instead of showing files one after the other, display them horizontally as tabs.

Screenshot from 2019-05-26 21-41-31

Please let me know if there is anything I need to change.

### CODE REVIEW

**Comment by lauthieb** [comment]
Wow @Andrei0872, thanks a lot for your PR!
I’ll review the code pretty soon. Thanks for this contribution!

**Comment by lauthieb** [comment]
Hello @Andrei0872,
Before going to the code, I have tested your feature in development mode.
Here is my remarks: 
- When I create a note with only 1 file, it would be nice if we can save it without click on "Add a file" button.
- The tabs on top are confusing, maybe this tab bar should go at the bottom of a note...
- You have a UI bug when we click on a tab, check this GIF: 
May-27-2019 14-16-05

Thanks a lot for your contribution. If you can fix my remarks, I will review the code deeply and sure this PR could be merged ;)

**Comment by Andrei0872** [comment]
Thank you for the feedback, @lauthieb! 
I’ll fix the remarks as soon as possible.

**Comment by Andrei0872** [comment]
Hello, @lauthieb ! I just sent my attempt to fix the remarks.
I couldn't reproduce the 3rd one since the tab bar is now on the bottom.

File: src/renderer/components/modals/create-note-modal/CreateNoteModal.html [hunk 1]
```diff
@@ -7,27 +7,17 @@ <h2 class="text-red" v-if="displayDupError">Oh sorry, you can't have duplicated
 
     <section class="modal-card-body">
       <b-field v-if="!gistsSelected" horizontal label="Name">
-        <b-input
-          type="text"
-          ref="noteName"
-          v-model="note.name"
-          placeholder="Your note name">
+        <b-input type="text" ref="noteName" v-model="note.name" placeholder="Your note name">
```
Comment (lauthieb) [inline src/renderer/components/modals/create-note-modal/CreateNoteModal.html]: I think there's a formatting problem there

Comment (lauthieb) [reply src/renderer/components/modals/create-note-modal/CreateNoteModal.html]: Do you follow eslint config file in your editor configuration?

**Comment by Andrei0872** [comment]
Hi, @lauthieb!
I couldn’t reproduce the error since in the gif the tabs were placed at the top of the card and now they are at the bottom.

I’m not sure I followed all the rules there, but I’ll have a look at it tomorrow:)

File: src/renderer/components/modals/create-note-modal/CreateNoteModal.html (hunk 1 above)
Comment (Andrei0872) [reply src/renderer/components/modals/create-note-modal/CreateNoteModal.html]: Hi again, @lauthieb. I've spent some time trying stuff and I couldn't reproduce the bug.

Also, I ran `yarn eslint` & `yarn eslint:fix` and nothing got fixed. What should I do now?

**Comment by lauthieb** [comment]
Hi @Andrei0872, can you fix conflicts by removing the yarn.lock please?

**Comment by lauthieb** [comment]
I think I will merge pretty soon this #93 Pull Request to correct the actual bug (#92), then I will go through your Pull Request to merge this.
Sorry again for the delay... Ah.. Open Source :)

**Comment by Andrei0872** [comment]
Hi @lauthieb, I fixed the conflicts and hopefully everything should be working fine!

**Comment by lauthieb** [comment]
Thanks @Andrei0872 !
Did you try to reinstall your node modules?
It seems your development doesn't work when `npm run dev`:
Capture d’écran 2019-10-10 à 23 53 56
Capture d’écran 2019-10-10 à 23 54 11

**Comment by Andrei0872** [comment]
Hi @lauthieb , just pushed the fix!

**Comment by lauthieb** [comment]
On `master` branch (margin between cards):

Capture d’écran 2019-10-11 à 22 23 16

On your branch: 
Capture d’écran 2019-10-11 à 22 22 49

**Comment by lauthieb** [comment]
I always have this UI bug when I create/update a file (when I click on the button to display a file) even if tabs are in bottom.

Oct-11-2019 22-26-01

**Comment by Andrei0872** [comment]
Hmm.. that's odd(regarding the first issue). I have rebased my branch onto master(this master) and I can't explain why there is no margin between notes :thinking:.

Also, I couldn't reproduce the bug..
ezgif com-video-to-gif (10)

**Comment by lauthieb** [comment]
@Andrei0872 Can you please fix conflicts and update your branch with just your modifications?
Sorry for the inconvenience it was mandatory to force push master after several tries to fix this important bug #110

Here you can download the new version 1.2.3 : https://github.com/lauthieb/code-notes/releases/tag/1.2.3

Thanks a lot in advance for this.

**Comment by lauthieb** [comment]
Hi @Andrei0872, happy new year :tada: 
What about the conflict resolution? Do you want to continue this Pull Request?
Thanks a in advance. Have a great day :)

**Comment by Andrei0872** [comment]
Hi @lauthieb, Happy New Year!
Sorry, I didn't notice the previous comment. Sure, I'll have look. Hopefully, I'll get it done by the evening.

Thank you!

**Comment by lauthieb** [comment]
Thanks @Andrei0872, take time if needed, no problem.
And feel free if you have any question.

**Comment by Andrei0872** [comment]
Hi @lauthieb , I fixed the conflicts, but when I run `npm run dev`, I get this error:

```
const { Template, util: { createHash } } = _webpack2.default;
                          ^
TypeError: Cannot destructure property `createHash` of 'undefined' or 'null'.
```
I might happen due to webpack's current version?

I took a look at [this](https://github.com/webpack-contrib/mini-css-extract-plugin/issues/69) issue.

**Comment by lauthieb** [comment]
Hi @Andrei0872, are you sure? I see always 
image
on your PR

**Comment by Andrei0872** [comment]
Hmm. ~9k are from `yarn-error.log` which I guess it should be deleted. The other ~6.4k are from `package-lock.json`.
What should I do?

**Comment by Akoneto9** [comment]
Good work

**Comment by lauthieb** [comment]
Hi @Andrei0872,
I've just marked your Pull Request as closed because it has not had recent activity. 
It will be closed if no further activity occurs. 
Feel free to reopen it if you have time to contribute and fix conflicts... I haven't got any time to help you on this sorry...
Thanks again for your contributions.

**Pull request closed without merging**